src/engine/renderer/glsl_source/computeLight_fp.glsl

```diff
@@ -21,6 +21,19 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 // computeLight_fp.glsl - Light computing helper functions
 
+#if defined(r_specularMapping) || defined(USE_PHYSICAL_SHADING)
+uniform vec2            u_SpecularExponent;
+#define getSpecular() texture2D(u_SpecularMap, var_TexSpecular + texOffset)
```

Comment:
It would be better to take the coordinate as an argument instead of hard-coding the name of some local variable.

Reply:
this is to not have to put ifdefs in all files that does specular mapping to pass or not the variables… :-/

The clutter to add the ifdef to pass the variable or not is the same to do this compute or not :-/

Reply:
It's OK to use a nonexistent variable in the macro argument as long as it does not appear in the final preprocessed output. This is possible since preprocessing is a conceptually separate step from compiling.

Reply:
The thing is that if the function expects a parameter, you have to set this parameter in the calling code and if this parameter does not exist anytime, you have to put ifdef around every call in every file calling this function.

We know this functions is called by every code doing specular stuff _but_ the liquid shader. If there was not such exception there would be no reason to declare the stuff in the calling code, only in the common library.

There is another alternative I tried and it works but I don't know if it's clean enough: it would be to define an useless `var_TexSpecular` in `liquid_vp.glsl` so `computeLight_fp.glsl` is able to compile while prepended to `liquid_fp.glsl`. The code would not be run in any way, the symbol just needs to be defined so the whole file compile including code that's not used.

Reply:
What I'm saying is you can do something like this:
```
#ifdef USE_FOO
#define foo(foo_factor) get_foo(foo_factor, 3)
#else
#define foo(foo_factor) 1.0
#endif

...

#ifdef USE_FOO
float foo_factor = 2;
#endif

...

float x = foo(foo_factor);
```

Reply:
That can't help because in the liquid case, `foo_factor` does not exist while `USE_FOO` is defined.

The liquid shader use the exact same specular code but don't get specular map from files.
So we would be able to define both `USE_SPECULAR_MAPPING` and `WITHOUT_SPECULAR_MAP` for it but it would defeat the macro count reduction.

Reply:
I removed this commit refactoring specular stuff since I may have some other PR to merge before this refactoring to make life easier.